### PR TITLE
Jetpack Disconnect: Add 'Other' reason, small fixes

### PR DESCRIPTION
**Pull request by ockham**
Addresses @rachelmcr's feedback at p7rd6c-15q-p2 #comment-1787.

* Reword 'This plan didn’t include what I needed' to 'I was missing a feature'.
* Add 'Other' reason.

Furthermore:
* Remove unused `disconnect-site__survey-one` `className`s (grep to verify).

image

### CODE REVIEW

File: client/my-sites/site-settings/disconnect-site/confirm.jsx [hunk 1]
```diff
@@ -61,7 +62,7 @@ class ConfirmDisconnection extends PureComponent {
 	render() {
 		const { reason, siteId, siteSlug, translate } = this.props;
 		const previousStep = find(
-			without( keys( this.constructor.reasonWhitelist ), 'troubleshooting' ), // There's no step for Troubleshooting
+			without( keys( this.constructor.reasonWhitelist ), 'troubleshooting', 'other' ), // Redirect those back to initial survey
```
Comment (sirreal) [inline client/my-sites/site-settings/disconnect-site/confirm.jsx]: I find this `without keys` to be a bit hard to understand. Do you think the logic would be clearer in the `find` predicate? I think the following is synonymous:

```js
find(
  this.constructor.reasonWhitelist,
  r => ! includes( [ 'troubleshooting', 'other' ], r ) && r === reason,
)
```

File: client/my-sites/site-settings/disconnect-site/disconnect-survey.jsx [hunk 2]
```diff
@@ -27,32 +27,21 @@ const DisconnectSurvey = ( { confirmHref, isPaidPlan, siteId, siteSlug, translat
...
-				href={ `/settings/disconnect-site/too-expensive/${ siteSlug }` }
-				className="disconnect-site__survey-one"
-			>
+			<CompactCard href={ `/settings/disconnect-site/too-expensive/${ siteSlug }` }>
 				{ translate( 'This plan is too expensive' ) }
 			</CompactCard>
 		) }
-		<CompactCard
-			href={ confirmHref + '?reason=troubleshooting' }
-			className="disconnect-site__survey-one"
```
Comment (sirreal) [inline client/my-sites/site-settings/disconnect-site/disconnect-survey.jsx]: ❤️

File: client/my-sites/site-settings/disconnect-site/confirm.jsx (hunk 1 above)
Comment (tyxla) [reply client/my-sites/site-settings/disconnect-site/confirm.jsx]: I kinda prefer Bernie's version to this one 😉  Seems more readable to me.

Comment (sirreal) [reply client/my-sites/site-settings/disconnect-site/confirm.jsx]: 👍  fair enough, retracted 🙂

**Comment by ockham** [comment]
> Also, the following sounds a bit strange to me:

>> I was missing a feature

Indeed. :slightly_frowning_face: 

> Maybe @Automattic/editorial could have a look to give suggestions? The plan is missing a feature, and I wish the plan had it. Tagged for copy review.

Alas, the previous (plan-centric) wording (`This plan didn’t include what I needed`) didn't apply to the free "plan", as @rachelmcr pointed out in the comment that I linked to:

> Also, I realized looking at these options right now that the option “This plan didn’t include what I needed” sounds a bit funny to me when I’m going through the survey on the free plan. If I were using Jetpack for free I’m not sure I’d think of myself as being on a plan.

**Comment by kristastevens** [comment]
How about: 

* A feature I need wasn't in the plan. 

I think that's clear and plain. Let me know what you think.

**Comment by ockham** [comment]
Hmm, that mentions _plan_ again tho... It was Rachel's comment

> Also, I realized looking at these options right now that the option “This plan didn’t include what I needed” sounds a bit funny to me when I’m going through the survey on the free plan. If I were using Jetpack for free I’m not sure I’d think of myself as being on a plan.

that initially prompted me to try and avoid mentioning _plan_...

**Comment by kristastevens** [comment]
ere's an alternate suggestion: 

* A feature I need was missing.

**Pull request merged**